Hpc/Ldp - Guide update LDP in managed vSphere (#6790)
ovh/docs: pages/hosted_private_cloud/hosted_private_cloud_powered_by_vmware/vmware_ldp/guide.fr-fr.md

---

> title: "Logs Data Platform - Collection les logs VMware on OVHcloud"

Reviewer: ```suggestion
title: "Logs Data Platform - Collecter les logs VMware on OVHcloud"
```

excerpt: "Découvrez comment souscrire un abonnement et activer le log forwarder afin de transférer les journaux VMware on OVHcloud vers un stream Logs Data Platform"
updated: 2024-09-16
---

> [!primary]

> > Cette fonctionnalité est actuellement en phase Bêta. Les infrastructures Logs Data Platform ne propose pas à ce jour les mêmes niveaux de qualifications tel que SecNumCloud, PCI-DSS, HDS qu'avec un environnement VMware on OVHcloud, cependant l'activation est toujours possible.

Reviewer: ```suggestion
> Cette fonctionnalité est actuellement en phase Bêta. À date, les infrastructures Logs Data Platform ne proposent pas les mêmes niveaux de qualifications (tels que SecNumCloud, PCI-DSS, HDS) qu'avec un environnement VMware on OVHcloud. L'activation est cependant toujours possible.
```

> 

## Objectif

**L'objectif est de vous montrer comment collecter les logs VMware vSphere on OVHcloud dans un stream Logs Data Platform.**

## Prérequis

- Disposer d'un [compte client OVHcloud](/links/manager).
- Disposer d'une offre VMware vSphere on OVHcloud.
- Avoir suivi le guide « [Introduction à Logs Data Platform](/pages/manage_and_operate/observability/logs_data_platform/getting_started_introduction_to_LDP) ».
- Vous devez avoir le `logForwarder` activé. Pour le vérifier, exécutez [l'appel API suivant](#security-options).
- Un stream Logs Data Platform actif, afin de lancer la souscription au sein de votre VMware vSphere managé on OVHcloud.

## En pratique

> [!primary]
> Prenez en compte que l'activation du **transfert de logs (Log Forwarding)** est gratuite.
>

> > Cependant, vous devez pour assurer le bon fonctionnement de votre cluster dédié Logs Data Platform, des charges de stockage peuvent s'appliquer avec le temps selon le tarif standard. Ainsi qu'avec l'utilisation du catalogue de services managés (Graylogs, OpenSearch, Logstash etc..).

Reviewer: ```suggestion
> Cependant, pour assurer le bon fonctionnement de votre cluster dédié Logs Data Platform, des frais liés au stockage ainsi qu'à l'utilisation du catalogue de services managés (Graylogs, OpenSearch, Logstash etc..) peuvent s'appliquer selon le tarif standard.
```

Un kind est un « type » de logs que votre produit génère.

Ils sont les types de logs que vous voulez transférer à votre stream Logs Data Platform. Voici des exemples qui peuvent être disponibles en fonction des composants de votre architecture VMware on OVHcloud :

- `esxi` : Seulement certaines applications sont redirigées.
- `nsxtEdge` : Tout est redirigé, pas de filtre.
- `vcsa` : Filtré par application.
- `nsxtManager` : Filtré par application.

### Étape 1 - Activer le Log Forwarder dans un VMware vSphere on OVHcloud

> [!warning]
> Si le pack `logForwarder` n'est pas activé au sein de votre pack de base d'options (base ou advanced security), [contactez le support OVHcloud](https://help.ovhcloud.com/csm?id=csm_get_help).
> 

> > Si vous voulez la fonctionnalité `logForwarder` sans les packs de base et de sécurité avancées, [contactez le support OVHcloud](https://help.ovhcloud.com/csm?id=csm_get_help) pour l'activer manuellement.

Reviewer: ```suggestion
> Si vous souhaitez utiliser la fonctionnalité `logForwarder` sans les packs de base et de sécurité avancée, [contactez le support OVHcloud](https://help.ovhcloud.com/csm?id=csm_get_help) pour l'activer manuellement.
```

>

#### Via l'API OVHcloud <a name="security-options"></a>

> [!primary]

> > Consultez le guide [Premiers pas avec les API OVHcloud](/pages/manage_and_operate/api/first-steps) pour vous familiariser avec l'utilisation des APIv6 OVHcloud.

Reviewer: ```suggestion
> Consultez le guide « [Premiers pas avec les API OVHcloud](/pages/manage_and_operate/api/first-steps) » pour vous familiariser avec l'utilisation des APIv6 OVHcloud.
```

>

> **Référencement de tous les appels API les packs de sécurités** :

Reviewer: ```suggestion
**Référencement de tous les appels API des packs de sécurités** :
```

Pour vérifier les options exigées pour permettre le fonctionnement de la fonctionnalité `logForwarder` au sein de votre VMware vSphere managé on OVHcloud, exécutez l'appel API suivant :

> [!api]
>
> @api {v1} /dedicatedCloud GET /dedicatedCloud/{serviceName}/securityOptions/compatibilityMatrix
>

> Laissez vides les 2 champs booléens suivant`showIncompatible` et `showInternal` disponibles.

Reviewer: ```suggestion
Laissez vides les 2 champs booléens `showIncompatible` et `showInternal` disponibles.
```

]
```

### Étape 2 - Création d'un stream Log Data Platform

> [!primary]
> Les ressources VMware on OVHcloud et Logs Data Platform doivent bien appartenir au même compte OVHcloud.
>

> > Vos journaux sont manipulés à des fins de sécurité et d'observabilité dans les clusters privés Logs Data Platform. Pour plus d'informations sur la tarification LDP, consultez la page LDP de [ce lien](/links/manage-operate/ldp).

Reviewer: ```suggestion
> Vos journaux sont manipulés à des fins de sécurité et d'observabilité dans les clusters privés Logs Data Platform. Pour plus d'informations sur la tarification LDP, consultez la page LDP disponible via [ce lien](/links/manage-operate/ldp).
```

>

Pour créer une souscription, un stream est nécessaire. Vous pouvez créer un stream temporaire afin de souscrire votre abonnement avec le `streamId` et l'appel API POST de l'étape 3.

#### Via l'espace client OVHcloud

> Vous pouvez vous référer à ce guide pour retrouver comment créer un stream depuis l'interface Logs Data Platform : « [Quick start for Logs Data Platform (EN)](/pages/manage_and_operate/observability/logs_data_platform/getting_started_quick_start) »

Reviewer: ```suggestion
Vous pouvez vous référer à ce guide pour retrouver comment créer un stream depuis l'interface Logs Data Platform : « [Quick start for Logs Data Platform (EN)](/pages/manage_and_operate/observability/logs_data_platform/getting_started_quick_start) ».
```

#### Via l'API OVHcloud

> Pour la création d'un stream, un `serviceName` LDP est nécessaire. Ce service Name fait reference à un ID unique de votre compte LDP, e.g. (`ldp-xx-XXXXX`)

Reviewer: ```suggestion
Pour la création d'un stream, un `serviceName` LDP est nécessaire. Ce `serviceName` fait reference à un ID unique de votre compte LDP (par exemple `ldp-xx-XXXXX`).
```

Vous devez donc préalablement avoir un compte LDP temporaire pour la création d'un stream temporaire.

Voici l'appel API de création d'un stream :

> [!api]
>
> @api {v1} /dbaas/logs POST /dbaas/logs/{serviceName}/output/graylog/stream
>

> **Paramètres** :
>
> - `description` : Description de votre stream LDP.
> - `title`: Titre de votre stream LDP.
> 

Retour :

```json
{
  "aliasId": null,
  "createdAt": "2024-08-28T12:25:13.544323+02:00",
  "dashboardId": null,
  "encryptionKeyId": null,
  "indexId": null,
  "inputId": null,
  "operationId": "e34f7a06-386b-4602-8d16-d45249197d40",
  "osdId": null,
  "roleId": null,
  "serviceName": "ldp-pu-66281",
  "state": "RUNNING",
  "streamId": null,
  "subscriptionId": null,
  "tokenId": null,
  "updatedAt": "2024-08-28T12:25:13.554173+02:00"
}
```

Récupérez le `streamId` et sauvegardez-le (copier-coller) dans un éditeur de texte. Vous en aurez besoin pour activer votre souscription VMware on OVHcloud avec le stream temporaire Logs Data Platform en question.

> Nous allons voir dans l'étape suivante comment souscrire votre abonnement vSphere managé à un stream LDP temporaire.

Reviewer: ```suggestion
Nous allons voir dans l'étape suivante comment attacher votre abonnement vSphere managé à un stream LDP temporaire.
```
Not sure about this. But "souscrire quelque chose à" is not right here.

### Étape 3 - Souscription de l'abonnement Logs Data Platform

> [!warning]

> > Pour disposer d'une souscription Logs Data Platform actif avec VMware on OVHcloud, vous devez avoir un stream actif. 

Reviewer: ```suggestion
> Pour disposer d'une souscription Logs Data Platform active avec VMware on OVHcloud, vous devez avoir un stream actif. 
```

> 

> > Si vous voulez avoir un stream sur le même compte et avec le même niveau de sécurité que votre environnement VMware on OVHcloud, vous devez faire le transfert de ce stream à celui de votre infrastructure privée (à ce jour). La responsabilité du transfert vous incombe, ainsi que du niveau de sécurité que vous voulez.

Reviewer: ```suggestion
> À date, si vous voulez avoir un stream sur le même compte et avec le même niveau de sécurité que votre environnement VMware on OVHcloud, vous devez faire le transfert de ce stream à celui de votre infrastructure privée. La responsabilité du transfert vous incombe, ainsi que du niveau de sécurité que vous voulez.
```

> 

> > Vous pouvez vous referer à la [documentation Log Data Platform](/pages/manage_and_operate/observability/logs_data_platform/getting_started_responsibility_model), si vous voulez plus d'information sur les niveaux de sécurité et de responsabilités disponibles.

Reviewer: ```suggestion
> Vous pouvez vous référer à la [documentation Logs Data Platform](/pages/manage_and_operate/observability/logs_data_platform/getting_started_responsibility_model) pour plus d'information sur les niveaux de sécurité et de responsabilités disponibles.
```

> 

#### Via l'espace client OVHcloud

Cette fonctionnalité n'est pas encore disponible dans la section VMware on OVHcloud de l'espace client.

#### Via l’API OVHcloud <a name="activation"></a>

Utilisez les appels API suivants pour établir la liste des abonnements de votre compte VMware on OVHcloud.

> **Référencement des appels API**

Reviewer: ```suggestion
**Référencement des appels API :**
```

| **Méthode** | **Chemin**                                                     | **Description**                                |
|:-----------:|:---------------------------------------------------------------|:-----------------------------------------------|
|     GET     | /dedicatedCloud/{serviceName}/log/kind                         | Types de logs pour votre service               |
|     GET     | /dedicatedCloud/{serviceName}/log/kind/{name}                  | Obtenir les propriétés de cet objet            |
|     GET     | /dedicatedCloud/{serviceName}/log/subscription                 | Inscrivez-vous pour votre service              |

> |    POST     | /dedicatedCloud/{serviceName}/log/subscription                 | Créez un abonnement log pour votre service     |

Reviewer: ```suggestion
|    POST     | /dedicatedCloud/{serviceName}/log/subscription                 | Créez un abonnement LDP pour votre service     |
```

|     GET     | /dedicatedCloud/{serviceName}/log/subscription                 | Obtenir les propriétés de cet objet            |

> |   DELETE    | /dedicatedCloud/{serviceName}/log/subscription                 | Supprimer un abonnement log pour votre service |

Reviewer: ```suggestion
|   DELETE    | /dedicatedCloud/{serviceName}/log/subscription                 | Supprimer un abonnement LDP pour votre service |
```

> Pour récupérer le **streamId** de votre compte LDP, consultez le guide « [Premiers pas Logs Data Platform](/pages/manage_and_operate/observability/logs_data_platform/getting_started_quick_start) ».

Reviewer: ```suggestion
Pour récupérer le **streamId** de votre compte LDP, consultez le guide « [Quick start for Logs Data Platform (EN)](/pages/manage_and_operate/observability/logs_data_platform/getting_started_quick_start) ».
```

> [!api]
> 
> @api {v1} /dedicatedCloud POST /dedicatedCloud/{serviceName}/log/subscription
>
>
> **Paramètres** :
>

> > - `serviceName` : Nom du service vSphere managé, e.g `pcc-XXX-XXX-XXX-XXX`.

Reviewer: ```suggestion
> - `serviceName` : Nom du service vSphere managé, (ex : `pcc-XXX-XXX-XXX-XXX`).
```

> > - `kind` : Kind VMware que le forwarder utilise, e.g (Disponible : `nsxtEdge ┃ vcsa ┃ nsxtManager ┃ esxi`).

Reviewer: ```suggestion
> - `kind` : Kind VMware que le forwarder utilise (ex : Disponible : `nsxtEdge ┃ vcsa ┃ nsxtManager ┃ esxi`).
```

> > - `streamId` : Identifiant du flux (stream) de destination, e.g (uuid : `ggb8d894-c491-433e-9c87-50a8bf6fe773`).

Reviewer: ```suggestion
> - `streamId` : Identifiant du flux (stream) de destination (ex : uuid : `ggb8d894-c491-433e-9c87-50a8bf6fe773`).
```

>

Exemple de paramètres en input pour l'appel API :

```json
{
  "serviceName": "pcc-XXX-XXX-XXX-XXX",
  "kind": "esxi",
  "streamId": "Null",
}
```

La requête GET permet de lister les stream ID au sein de votre souscription.

### Étape 4 - Administrer vos logs VMware on OVHcloud

> [!primary]

> > Si vous voulez connaitre tous les services et fonctionnalités que propose Log Data Plaform, consultez la page OVHcloud produit de [ce lien](/links/manage-operate/ldp).

Reviewer: ```suggestion
> Si vous voulez connaitre tous les services et fonctionnalités que propose Log Data Plaform, consultez la page dédiée via [ce lien](/links/manage-operate/ldp).
```

> 

> L'administration de vos streams LDP peuvent être fait depuis l'API OVHcloud, depuis l'espace client Bare Metal Logs Data Platform, depuis l'UI Graylog, depuis l'UI OpenSearch et bientôt dans l'espace client.

Reviewer: ```suggestion
L'administration de vos streams LDP peut être faite depuis l'API OVHcloud, depuis la partie `Bare Metal`{.action} >  `Logs Data Platform`{.action} de l'espace client OVHcloud, depuis l'UI Graylog, depuis l'UI OpenSearch.
```

#### Via l'espace client OVHcloud

> Vous pouvez vous référer à ce guide pour retrouver comment administrer vos flux (streams) depuis le control panel Log Data Platform : « [Quick start for Logs Data Platform (EN)](/pages/manage_and_operate/observability/logs_data_platform/getting_started_quick_start) » (EN).

Reviewer: ```suggestion
Vous pouvez vous référer à ce guide pour retrouver comment administrer vos flux (streams) depuis la section `Log Data Platform`{.action} de l'espace client OVHcloud : « [Quick start for Logs Data Platform (EN)](/pages/manage_and_operate/observability/logs_data_platform/getting_started_quick_start) ».
```

> **Paramètres** :
>

> > - `serviceName` : Nom de service de votre vSphere managé, e.g `pcc-XXX-XXX-XXX-XXX`.

Reviewer: ```suggestion
> - `serviceName` : Nom de service de votre vSphere managé (ex : `pcc-XXX-XXX-XXX-XXX`).
```

> > - `kind` : Nom kind VMware que le forwarder utilise, e.g (`nsxtEdge ┃ vcsa ┃ nsxtManager ┃ esxi`).

Reviewer: ```suggestion
> - `kind` : Nom du kind VMware que le forwarder utilise (ex : `nsxtEdge ┃ vcsa ┃ nsxtManager ┃ esxi`).
```

>

**Comment lister vos Kind disponibles** ?

> [!api]
>
> @api {v1} /dedicatedCloud GET /dedicatedCloud/{serviceName}/log/kind/{name}
>

> **Paramètres** :
>

> > - `name` : Nom kind VMware que le forwarder utilise, e.g (Disponible : `nsxtEdge ┃ vcsa ┃ nsxtManager ┃ esxi`).

Reviewer: ```suggestion
> - `name` : Nom du kind VMware que le forwarder utilise (ex : Disponible : `nsxtEdge ┃ vcsa ┃ nsxtManager ┃ esxi`).
```

> > - `serviceName` : Nom de service de votre vSphere managé, e.g `pcc-XXX-XXX-XXX-XXX`.

Reviewer: ```suggestion
> - `serviceName` : Nom de service de votre vSphere managé (ex : `pcc-XXX-XXX-XXX-XXX`).
```

> 

Exemples de retours pour différents Kind :

**Kind name : NSX-T Edge**

```json
{
  "name": "nsxtEdge",
  "createdAt": "2024-05-31T22:17:40+02:00",
  "kindId": "Null",
  "displayName": "NSX-T Edges",
  "additionalReturnedFields": [
    "priority",
    "application_name",
    "level",
    "comp",
    "s2comp",
    "subcomp"
  ],
  "updatedAt": "2024-05-31T22:17:40+02:00"
}
```

**Kind name : NSX-T Manager**

```json

> {

Reviewer: ```suggestion
```

{
  "displayName": "NSX-T Manager",
  "kindId": "Null",
  "additionalReturnedFields": [
    "priority",
    "application_name",
    "level",
    "comp",
    "s2comp",
    "subcomp"
  ],
  "createdAt": "2024-05-31T22:17:40+02:00",
  "updatedAt": "2024-05-31T22:17:40+02:00",
  "name": "nsxtManager"
}
```

**Kind name : VCSA**

```json
{
  "name": "vcsa",
  "createdAt": "2024-05-31T22:17:40+02:00",
  "additionalReturnedFields": [
    "priority",
    "application_name",
    "level"
  ],
  "updatedAt": "2024-05-31T22:17:40+02:00",
  "kindId": "Null",
  "displayName": "VCSA"
}
```

**Kind name : ESXI**

```json
{
  "kindId": "Null",
  "displayName": "esxi",
  "name": "esxi",
  "createdAt": "2024-04-26T22:27:57+02:00",
  "updatedAt": "2024-04-26T22:27:57+02:00",
  "additionalReturnedFields": [
    "level",
    "application_name"
  ]
}
```

> ### Étape 5 - Désactiver votre abonnement de souscription Log Data Platform

Reviewer: ```suggestion
### Étape 5 - Désactiver votre abonnement de souscription Logs Data Platform
```

> [!primary]

> > La résiliation de votre abonnement Log Data Platform VMware on OVHcloud ne signifie pas la suppression de vos streams. Le stockage consommé au moment de la désactivation reste soumis à facturation.

Reviewer: ```suggestion
> La résiliation de votre abonnement Logs Data Platform VMware on OVHcloud ne signifie pas la suppression de vos streams. Le stockage consommé au moment de la désactivation reste soumis à facturation.
```

>

> > **Note** : Il n'est possible (à ce jour) que de supprimer un stream dans son intégralité (en entier) et non certains inputs granulaires.

Reviewer: ```suggestion
> **Note** : à date, seule la suppression intégrale d'un stream est possible.  Il n'est possible de supprimer uniquement certains inputs granulaires.
```

> **Paramètres** :
>

> > - `serviceName` : Nom de service de votre vSphere managé, e.g `pcc-XXX-XXX-XXX-XXX`.

Reviewer: ```suggestion
> - `serviceName` : Nom de service de votre vSphere managé (ex : `pcc-XXX-XXX-XXX-XXX`).
```

> > - `subscriptionId` : Identifiant de souscription de votre abonnement LDP, e.g `8e729f36-3e5d-4e08-9723-62b2b39c857a`.

Reviewer: ```suggestion
> - `subscriptionId` : Identifiant de souscription de votre abonnement LDP (ex :  `8e729f36-3e5d-4e08-9723-62b2b39c857a`).
```
